testsuite/tests/cpp-api/cpp_stubs3.c

```diff
@@ -0,0 +1,17 @@
+#define CAML_INTERNALS
```

Comment:
The CAML_INTERNALS sections of the OCaml 5 header files contain too much C code (as inline functions) to be compilable by a C++ compiler.  So, this 3rd test will never work, I'm afraid.  Let's just test without CAML_INTERNALS.